diff -up plugins/org.eclipse.ecf.core.identity/META-INF/MANIFEST.MF.qualifier plugins/org.eclipse.ecf.core.identity/META-INF/MANIFEST.MF
--- plugins/org.eclipse.ecf.core.identity/META-INF/MANIFEST.MF.qualifier	2008-07-07 15:54:07.000000000 -0400
+++ plugins/org.eclipse.ecf.core.identity/META-INF/MANIFEST.MF	2008-07-07 15:56:20.000000000 -0400
@@ -2,7 +2,7 @@ Manifest-Version: 1.0
 Bundle-ManifestVersion: 2
 Bundle-Name: %plugin.name
 Bundle-SymbolicName: org.eclipse.ecf.identity;singleton:=true
-Bundle-Version: 2.0.0.qualifier
+Bundle-Version: 2.0.0.v20080611-1715
 Bundle-Activator: org.eclipse.ecf.internal.core.identity.Activator
 Bundle-Localization: plugin
 Bundle-Vendor: %plugin.provider
diff -up plugins/org.eclipse.ecf.filetransfer/META-INF/MANIFEST.MF.qualifier plugins/org.eclipse.ecf.filetransfer/META-INF/MANIFEST.MF
--- plugins/org.eclipse.ecf.filetransfer/META-INF/MANIFEST.MF.qualifier	2008-07-07 15:56:01.000000000 -0400
+++ plugins/org.eclipse.ecf.filetransfer/META-INF/MANIFEST.MF	2008-07-07 15:56:20.000000000 -0400
@@ -2,7 +2,7 @@ Manifest-Version: 1.0
 Bundle-ManifestVersion: 2
 Bundle-Name: %plugin.name
 Bundle-SymbolicName: org.eclipse.ecf.filetransfer;singleton:=true
-Bundle-Version: 2.0.0.qualifier
+Bundle-Version: 2.0.0.v20080611-1715
 Bundle-Activator: org.eclipse.ecf.internal.filetransfer.Activator
 Bundle-Vendor: %plugin.provider
 Eclipse-LazyStart: true
diff -up plugins/org.eclipse.ecf/META-INF/MANIFEST.MF.qualifier plugins/org.eclipse.ecf/META-INF/MANIFEST.MF
--- plugins/org.eclipse.ecf/META-INF/MANIFEST.MF.qualifier	2008-07-07 15:56:00.000000000 -0400
+++ plugins/org.eclipse.ecf/META-INF/MANIFEST.MF	2008-07-07 15:56:20.000000000 -0400
@@ -1,7 +1,7 @@
 Manifest-Version: 1.0
 Bundle-Name: %plugin.name
 Bundle-SymbolicName: org.eclipse.ecf;singleton:=true
-Bundle-Version: 2.0.0.qualifier
+Bundle-Version: 2.0.0.v20080611-1715
 Bundle-Activator: org.eclipse.ecf.internal.core.ECFPlugin
 Bundle-Vendor: %plugin.provider
 Bundle-Localization: plugin
diff -up plugins/org.eclipse.ecf.provider.filetransfer/META-INF/MANIFEST.MF.qualifier plugins/org.eclipse.ecf.provider.filetransfer/META-INF/MANIFEST.MF
--- plugins/org.eclipse.ecf.provider.filetransfer/META-INF/MANIFEST.MF.qualifier	2008-07-07 15:56:01.000000000 -0400
+++ plugins/org.eclipse.ecf.provider.filetransfer/META-INF/MANIFEST.MF	2008-07-07 15:56:20.000000000 -0400
@@ -2,7 +2,7 @@ Manifest-Version: 1.0
 Bundle-ManifestVersion: 2
 Bundle-Name: %plugin.name
 Bundle-SymbolicName: org.eclipse.ecf.provider.filetransfer;singleton:=true
-Bundle-Version: 2.0.0.qualifier
+Bundle-Version: 2.0.0.v20080611-1715
 Bundle-Activator: org.eclipse.ecf.internal.provider.filetransfer.Activator
 Bundle-Vendor: %plugin.provider
 Bundle-Localization: plugin
diff -up plugins/org.eclipse.ecf.provider.filetransfer.ssl/META-INF/MANIFEST.MF.qualifier plugins/org.eclipse.ecf.provider.filetransfer.ssl/META-INF/MANIFEST.MF
--- plugins/org.eclipse.ecf.provider.filetransfer.ssl/META-INF/MANIFEST.MF.qualifier	2008-07-07 15:56:00.000000000 -0400
+++ plugins/org.eclipse.ecf.provider.filetransfer.ssl/META-INF/MANIFEST.MF	2008-07-07 15:56:20.000000000 -0400
@@ -2,7 +2,7 @@ Manifest-Version: 1.0
 Bundle-ManifestVersion: 2
 Bundle-Name: %plugin.name
 Bundle-SymbolicName: org.eclipse.ecf.provider.filetransfer.ssl
-Bundle-Version: 1.0.0.qualifier
+Bundle-Version: 1.0.0.v20080611-1715
 Bundle-Vendor: %plugin.provider
 Fragment-Host: org.eclipse.ecf.provider.filetransfer;bundle-version="2.0.0"
 Bundle-RequiredExecutionEnvironment: J2SE-1.4
diff -up plugins/org.eclipse.ecf.ssl/META-INF/MANIFEST.MF.qualifier plugins/org.eclipse.ecf.ssl/META-INF/MANIFEST.MF
--- plugins/org.eclipse.ecf.ssl/META-INF/MANIFEST.MF.qualifier	2008-07-07 15:56:01.000000000 -0400
+++ plugins/org.eclipse.ecf.ssl/META-INF/MANIFEST.MF	2008-07-07 15:56:20.000000000 -0400
@@ -2,7 +2,7 @@ Manifest-Version: 1.0
 Bundle-ManifestVersion: 2
 Bundle-Name: %plugin.name
 Bundle-SymbolicName: org.eclipse.ecf.ssl
-Bundle-Version: 1.0.0.qualifier
+Bundle-Version: 1.0.0.v20080611-1715
 Fragment-Host: org.eclipse.ecf;bundle-version="2.0.0"
 Bundle-RequiredExecutionEnvironment: J2SE-1.4
 Import-Package: org.eclipse.osgi.service.security;version="1.0.0",
