--- eclipse.platform.swt/bundles/org.eclipse.swt/Eclipse SWT PI/gtk/library/os.h.bak	2013-01-28 12:16:08.000000000 +0100
+++ eclipse.platform.swt/bundles/org.eclipse.swt/Eclipse SWT PI/gtk/library/os.h	2013-02-07 12:03:27.234528249 +0100
@@ -18,7 +18,7 @@
 
 #define NDEBUG
 
-#define G_DISABLE_DEPRECATED
+/*#define G_DISABLE_DEPRECATED*/
 #define GTK_DISABLE_SINGLE_INCLUDES
 /*
 #define GTK_DISABLE_DEPRECATED
